### PR TITLE
de-boiler-plate credential stripping logic

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
[test]

**Comment by bparees** [comment]
@jim-minter ptal

File: pkg/util/util.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	"github.com/docker/engine-api/types/container"
+)
+
+// SafeForLoggingContainerConfig returns a string version of the container.Config object
+// with sensitive information (proxy environment variables containing credentials)
+// redacted.
+func SafeForLoggingContainerConfig(config *container.Config) string {
+	origEnv := config.Env
+	strippedEnv := StripProxyCredentials(origEnv)
+	config.Env = strippedEnv
```
Comment (jim-minter) [inline pkg/util/util.go]: operating on the original config object and not a shallow copy could in principle expose us to a race in future - suggest copying the config object up front

File: pkg/util/util.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	s := fmt.Sprintf("%+v", config)
```
Comment (jim-minter) [inline pkg/util/util.go]: FWIW I think I'd envisaged returning a temporary copy and having the caller decide what they'd Sprintf, but this works too :)

**Comment by jim-minter** [comment]
I would be inclined to fix the comment around the potential race, otherwise lgtm.

File: pkg/util/util.go (hunk 2 above)
Comment (bparees) [inline pkg/util/util.go]: I figured there might be cases where it's easier to embed the logic of "what to not log" into the function itself and saves us the trouble of deep copying the object.

**Comment by bparees** [comment]
> I would be inclined to fix the comment around the potential race, otherwise lgtm.

@jim-minter updated, ptal.

**Comment by jim-minter** [comment]
lgtm

**Comment by bparees** [comment]
[merge]

**Pull request merged**